Unit and integration tests, and method corrections

Mefjuu94 opened the pull request:
* fixed view of pages: "map", "trip" on mobile devices.
* optimized imports, improved code readbility, removed unnecessary comments.
* added test to all database utils; "CommentsDAO", "CustomUserDAO", "TopDAO", "TripDAO", "UserRatingDAO", and "DateChecker" class.
* added function to:
- delete trip when I'am owner of the trip,
- return to homepagein from any page (exclude when create trip).

Mefjuu94 committed:
* optimized imports, improved code readability, removed unnecessary comments

Mefjuu94 committed:
* fixed docker connection issue, correctly configured values in hibernate.cfg file. Now it works.

Mefjuu94 committed:
* Added few test methods from CustomUserDAO. Two methods to tested left.

Mefjuu94 committed:
* Optimizing imports
* changed some fields And methods name to be more specific.
* written all tests using Docker to CustomUserDAO

Mefjuu94 committed:
* added test in CommentsDAOTests
* added email in index page to section "contact"

Mefjuu94 committed:
* added tests to CommentsDao
* added tests to TripDAO

Mefjuu94 committed:
* Added Tests to TripDAO (8/10)
* changed some not working methods.

Mefjuu94 committed:
* Added Tests rest of database utils.
* changed some methods to work properly

Mefjuu94 committed:
* Added UserRatingTests

Mefjuu94 committed:
* deleted unnecessary prints
* deleted DemoTestContainers.java
* added DateCheckerTest

Mefjuu94 committed:
* added "home button" to return to homepage from any subpage
* added "distance traveled" variable
* fixed "rate" page. Now data is update by dateChecker method
* modify updateUserStats - now can maintenance numberOfAnnouncements, distanceTraveled and numbersOfTrips

Mefjuu94 committed:
* update tests because of new variable of CustomUser

Mefjuu94 committed:
* update tests because of new variable of CustomUser

Mefjuu94 committed:
* move some code to MethodsHandler to clean code - method to rate Users
* added to participants owner of a trip
* added variable to know is some users to rate in UserPanel
* Results especially to search friends, now looks like announcements and have additional information, like number of trips, age ect...

**TODO map view to mobile devices!**

Mefjuu94 committed:
* adjust map view to mobile devices

Mefjuu94 committed:
* changed some properties in application.properties
* changed all database methods to use "criteria query" to properly close session created.
* customUser now have set automatically distance traveled to 0.00 when created
* DateChecker now fires after 10 minutes
* fixed view of map (create trip)
* make view to trip page

Mefjuu94 committed:
* Added function to delete trip (button)
* Removed button "return to homepage" from homepage
* fixed some issues in map view

Mefjuu94 committed:
* Added function to renew trip (the journey should have already been made)
* Changed all methods in DAO -> now sessions are closed automatically (try/catch with resources) or forced by finally block.
* added attributes to annotation (fetch = FetchType.EAGER) to avoid error of LazyInitializationException.

Mefjuu94 committed:
* Added css styles to renew trip.

Mefjuu94 committed:
* changed size of background images, now are lighter.
* hidden "renew button" in trip view when trip is active

Mefjuu94 committed:
* Added properly validation of character inpyt in  profile page
* deleted invalid statement in trip page (if group was "closed" then was impossible to join)
* deleted "locate me" button from map creator.
* deleted print in DateChecker
* added model in error controller to handle error "500" now if user write invalid username or password then return properly message
* added some properties to database config file (applicationProperties), because of error of max connection (too many clients already)

Mefjuu94 committed:
* now name of custom user is substring from email (characters before '@')
* added privacy Policy
* changed style in map, now only border is green when field is valid
* added hints when register new user
* deleted "trip id" from trip page

Mefjuu94 committed:
* Added full validation to register user with tooltips.
* pulled main function to rate users to index page in the middle between register and login tiles.

Mefjuu94 committed:
pictures

Mefjuu94 committed:
* changed styles in userPanelStyles.css, now it is in table form

Mefjuu94 committed:
* added better control of rating for users
* added "editUserChanges" in CustomUserDAO to take responsibility of "dateChecker" method. Because of that object are edited within one session.
* Rates are created automatically after check if trip is after date (in DateChecker method) with boolean variable who indicates if comment was done or not
* slimed down method "usersToRate" in "MethodHandler"
* Rebuild UserRating! now has 6 fields with connection to other entities.
- comment
- isFilled -> false if the reviewer did not rate the user (this allows us to know whether the review took place)
- rating -> it's just mark 1-5
- reviewer_id
- trip_id
- user_id

* added "getRate" method to UserRatingDAO
* Rebuild page to rate users (HTML,CSS,JS)

**Because of Rebuild UserRating Tests do not work - in progress!**

Mefjuu94 committed:
* Added all necessary test in UserRatingDAOTests
* added some methods to maintain session, and connection in database in every dao class
* Optimized imports

Mefjuu94 committed:
* fixed rate page - now selecting stars and the final rating is displayed correctly, and non-active elements are unclickable.
* fixed index page view, now "Login Grid Container" wraps and expands correctly depending on screen width.
* deleted comments

Mefjuu94 committed:
* added possibility to change password if user forget it. (passwordRetrieve.html)
* added information page (the same like Success page)- can take more specific information about what's happen.

blazejprzyluski commented on src/main/java/TTT/component/DateChecker.java:
```diff
@@ -1,30 +1,52 @@
 @Component
 public class DateChecker {
     TripDAO tripDAO = new TripDAO();
+    CustomUserDAO customUserDAO = new CustomUserDAO();
+    UserRatingDAO userRatingDAO = new UserRatingDAO();
 
```
private final

Mefjuu94 replied:
fixed

blazejprzyluski commented on src/main/java/TTT/component/DateChecker.java:
```diff
@@ -1,30 +1,52 @@
-                entity.setTripVisible(false);
-                tripDAO.updateTrip(entity.getId(),entity);
-                System.out.println("Trip id: " + entity.getId() + " will be no longer avaible because is after date");
+
+            if (entity.isTripVisible()) {
+                //load all participants of trip
+                List<CustomUser> participants = tripDAO.findTripID(entity.getId()).getParticipants();
+                CustomUser owner = entity.getOwner();
+                participants.add(owner); // add owner to list of people to change stats and create rate
+
+                double number = Double.parseDouble(entity.getDistanceOfTrip().substring(0, entity.getDistanceOfTrip().indexOf("km")));
+
+                if (entity.getTripDateTime().isBefore(LocalDateTime.now())) {
+                    customUserDAO.editUsersChanges(participants, number);
+                    entity.setTripVisible(false);
+
+                    for (int i = 0; i < participants.size(); i++) {
+                        for (int j = 0; j < participants.size(); j++) {
+                            if (participants.get(i) != participants.get(j)) {
+                                userRatingDAO.addRate(new UserRating(0, entity, false, "", participants.get(i), participants.get(j)));
+                            }
+                        }
+                    }
+                    tripDAO.updateTrip(entity);
+                }
```
Consider moving to trip dao

blazejprzyluski commented on src/main/java/TTT/controller/AnnouncementController.java:
```diff
@@ -30,33 +27,33 @@
 @Controller
 public class AnnouncementController {
 
     TripDAO tripDAO = new TripDAO();
     CommentsDAO commentsDAO = new CommentsDAO();
+    MethodsHandler methodsHandler = new MethodsHandler();
```
private final

Mefjuu94 replied:
fixed

blazejprzyluski commented on src/main/java/TTT/controller/MethodsHandler.java:
```diff
@@ -0,0 +1,83 @@
+
+public class MethodsHandler {
+
+    UserRatingDAO userRatingDAO = new UserRatingDAO();
```
private final

blazejprzyluski commented on src/main/java/TTT/controller/AnnouncementController.java:
```diff
@@ -30,33 +27,33 @@
     public String getAnnouncementsPage(Model model) {
         TripDAO tripDAO = new TripDAO();
-        ArrayList<Trip> rawtrips = new ArrayList<>(tripDAO.listAllAnnouncements());
+        ArrayList<Trip> rawTrips = new ArrayList<>(tripDAO.listAllAnnouncements());
```
List on the left

Mefjuu94 replied:
fixed

blazejprzyluski commented on src/main/java/TTT/controller/AnnouncementController.java:
```diff
@@ -30,33 +27,33 @@
 
     @GetMapping("/announcement")
     public String getAnnouncementsPage(Model model) {
         TripDAO tripDAO = new TripDAO();
```
new trip dao even though it's a field

blazejprzyluski commented on src/main/java/TTT/controller/AnnouncementController.java:
```diff
@@ -30,33 +27,33 @@
+        ArrayList<Trip> rawTrips = new ArrayList<>(tripDAO.listAllAnnouncements());
 
         LocalDateTime localDateTime = LocalDateTime.now();
         ArrayList<Trip> trips = new ArrayList<>();
```
List on the left

Mefjuu94 replied:
ok

blazejprzyluski commented on src/main/java/TTT/controller/AnnouncementController.java:
```diff
@@ -30,33 +27,33 @@
-                System.out.println("ten trip: " + rawtrips.get(i).getDestination() + " jest już odbyta!");
-            }else {
-                trips.add(rawtrips.get(i));
+        for (int i = 0; i < rawTrips.size(); i++) {
```
Would look better with enhanced for

Mefjuu94 replied:
done.

blazejprzyluski commented on src/main/java/TTT/controller/AnnouncementController.java:
```diff
@@ -65,19 +62,16 @@ public String getAnnouncementsPage(Model model) {
         List<String> tripsAndIDs = new ArrayList<>();
 
-        for (int i = 0; i < obejcts.size(); i++) {
+        for (Object[] obejct : obejcts) {
```
typo and unsafe cast to Object

Mefjuu94 replied:
changed!

blazejprzyluski commented on src/main/java/TTT/controller/AnnouncementController.java:
```diff
@@ -172,49 +166,33 @@ public String getAnyUserProfile(@PathVariable Long id, Model model) {
+    public String addMeToTrip(@RequestParam String tripId, @RequestParam String userId, Model model) {
 
         Trip trip = tripDAO.findTripID(Long.parseLong(tripId));
         CustomUserDAO customUserDAO = new CustomUserDAO();
```
another new dao

blazejprzyluski commented on src/main/java/TTT/controller/CustomUserController.java:
```diff
@@ -6,74 +6,52 @@
 
     CustomUserDAO customUserDAO = new CustomUserDAO();
     TripDAO tripDAO = new TripDAO();
+    MethodsHandler methodsHandler = new MethodsHandler();
```
private final, consider making static

blazejprzyluski commented on src/main/java/TTT/controller/CustomUserController.java:
```diff
@@ -6,74 +6,52 @@
-                tripsParticipated.add(trip);
-            }
-        }
+        List<Object[]> objects = tripDAO.listAllTripParticipantIds();
```
unsafe cast

blazejprzyluski commented on src/main/java/TTT/controller/MethodsHandler.java:
```diff
@@ -0,0 +1,83 @@
+                UserDetails userDetails = (UserDetails) principal;
+                return userDetails.getUsername(); // return name os user (email)
+            }
+            // if not return something
```
throw an exception

Mefjuu94 replied:
like that? I am not sure...

blazejprzyluski commented on src/main/java/TTT/controller/MethodsHandler.java:
```diff
@@ -0,0 +1,83 @@
+    }
+
+    protected ArrayList<Trip> listOfTrips(List<Object[]> objects, CustomUser customUser){
+        TripDAO tripDAO = new TripDAO();
```
another new dao instance

blazejprzyluski commented on src/main/java/TTT/controller/MethodsHandler.java:
```diff
@@ -0,0 +1,83 @@
+                Trip trip = tripDAO.findTripID(tripID);
+                tripsParticipated.add(trip);
+            }
+        }
```
duplicated code from CustomUserController

blazejprzyluski commented on src/main/java/TTT/controller/PasswordResetController.java:
```diff
@@ -0,0 +1,63 @@
+        customUser = customUserDAO.findCustomUserByEmail(email);
+        if (customUser != null) {
+            customUser.setPassword(newPassword);
+            customUserDAO.updateUserField(newPassword, customUser.getEmail(),"password");
```
Please add password verification

Mefjuu94 replied:
Done!

blazejprzyluski commented on src/main/java/TTT/controller/SecurityController.java:
```diff
@@ -4,55 +4,45 @@
     @GetMapping("/login")
-    public String getLoginPage(){
+    public String getLoginPage() {
         return "/";
```
rather use index here

Mefjuu94 replied:
ok!

blazejprzyluski commented on src/main/java/TTT/controller/SecurityController.java:
```diff
@@ -4,55 +4,45 @@
 
-    @RequestMapping(value="/logout", method = RequestMethod.GET)
-    public String logoutPage (HttpServletRequest request, HttpServletResponse response) {
+    @RequestMapping(value = "/logout", method = RequestMethod.GET)
```
Use Get mapping

Mefjuu94 replied:
done.

blazejprzyluski commented on src/main/java/TTT/databaseUtils/CommentsDAO.java:
```diff
@@ -14,12 +14,28 @@
+        this.sessionFactory = testSessionFactory;
+    }
+
+    public void close() {
+        if (sessionFactory != null) {
+            sessionFactory.close(); // Zamknięcie fabryki sesji
+        }
+    }
```
to be deleted

Mefjuu94 replied:
done

blazejprzyluski commented on src/main/java/TTT/databaseUtils/CommentsDAO.java:
```diff
@@ -104,18 +124,26 @@ public Comments findCommentID(long id) {
     }
 
-    public void editComment(long idOfComment, String commentString) {
+    public Boolean editComment(long idOfComment, String commentString) {
```
may return primitive

Mefjuu94 replied:
fixed

blazejprzyluski commented on src/main/java/TTT/databaseUtils/CustomUserDAO.java:
```diff
@@ -47,15 +60,47 @@ public boolean saveUser(CustomUser customUser) {
-    public List<CustomUser> listAllUsers() {
-        Session session = sessionFactory.openSession();
-        return session.createQuery("SELECT a FROM CustomUser a", CustomUser.class).getResultList();
+    public boolean editUsersChanges(List<CustomUser> customUsers, double number) {
```
change to updateUsersAfterTrip

Mefjuu94 replied:
done.

blazejprzyluski commented on src/main/java/TTT/databaseUtils/CustomUserDAO.java:
```diff
@@ -138,7 +182,7 @@ public boolean deleteCustomUser(String email) {
     }
 
-    public void updateUserTrips(String email, List<Trip> trips) {
+    public Boolean updateUserTrips(String email, List<Trip> trips) {
```
should be primitive

Mefjuu94 replied:
fixed

blazejprzyluski commented on src/main/java/TTT/databaseUtils/CustomUserDAO.java:
```diff
@@ -148,25 +192,29 @@ public void updateUserTrips(String email, List<Trip> trips) {
     }
 
-    public void updateUserField(String value, String email,String field) {
+    public Boolean updateUserField(String value, String email, String field) {
```
primitive

Mefjuu94 replied:
fixed

blazejprzyluski commented on src/main/java/TTT/databaseUtils/CustomUserDAO.java:
```diff
@@ -176,23 +224,34 @@ public void updateUserField(String value, String email,String field) {
     }
 
-    public void updateUserAge(String email, int age) {
+    public Boolean updateUserAge(String email, int age) {
```
primitive

Mefjuu94 replied:
fixed

blazejprzyluski commented on src/main/java/TTT/databaseUtils/CustomUserDAO.java:
```diff
@@ -176,23 +224,34 @@ public void updateUserField(String value, String email,String field) {
-    public void updateUserAge(String email, int age) {
+    public Boolean updateUserAge(String email, int age) {
         Transaction transaction = null;
+        if (age > 120) {
```
Check if age can be negative

Mefjuu94 replied:
ok

blazejprzyluski commented on src/main/java/TTT/databaseUtils/TripDAO.java:
```diff
@@ -16,16 +16,34 @@
+
+    public TripDAO() {
+    }
+
+    public void close() {
+        if (sessionFactory != null) {
+            sessionFactory.close(); // Zamknięcie fabryki sesji
+        }
+    }
```
toBeDeleted

Mefjuu94 replied:
done.

blazejprzyluski commented on src/main/java/TTT/databaseUtils/TripDAO.java:
```diff
@@ -16,16 +16,34 @@
             session.merge(trip);
             transaction.commit();
             System.out.println("The trip has been created!");
+            session.close();
```
unnecessary close

Mefjuu94 replied:
yes...

blazejprzyluski commented on src/main/java/TTT/databaseUtils/TripDAO.java:
```diff
@@ -124,29 +150,35 @@ public Trip findTripID(long id) {
-        Session session = sessionFactory.openSession();
-        return session.createNativeQuery("SELECT trip_id, user_id FROM trip_participants", Object[].class)
-                .getResultList();
+        Session session = null;
+        try {
+            session = sessionFactory.openSession();
+            return session.createNativeQuery("SELECT trip_id, user_id FROM trip_participants", Object[].class)
+                    .getResultList();
+        }finally {
+            assert session != null;
+            session.close();
+        }
```
try with resources

blazejprzyluski commented on src/main/java/TTT/databaseUtils/TripDAO.java:
```diff
@@ -124,29 +150,35 @@ public Trip findTripID(long id) {
     }
 
-    public void updateTrip(long idTrip, Trip trip) {
+    public Boolean updateTrip(Trip trip) {
```
primitive

blazejprzyluski commented on src/main/java/TTT/databaseUtils/UserRatingDAO.java:
```diff
@@ -1,75 +1,147 @@
+        }
+    }
+
+    public List<UserRating> listAllRates(long userId) {
+        try (Session session = sessionFactory.openSession()) {
+            session.clear();
+            String sql = "SELECT * FROM userrating WHERE user_id = :userId AND isFilled = true";
+            return session.createNativeQuery(sql, UserRating.class)
+                    .setParameter("userId", userId)
+                    .getResultList();
+        }
+    }
```
to be deleted

blazejprzyluski commented on src/main/java/TTT/security/SecurityConfig.java:
```diff
@@ -32,28 +27,28 @@ public SecurityConfig(CustomUserSecurityService securityService) {
-                                "/trips/","/addComment","/addMe","/announcement","/findFriend","/searchResults",
-                                "/tripsOwned","/tripsParticipated","resources/**","/downloadGpx","/delete_participant",
-                                "/findTrip","/myProfile","/addRate",
+                        .requestMatchers("/map", "/sendData", "/logout", "/userProfile/", "/updateField",
+                                "/trips/", "/addComment", "/addMe", "/announcement", "/findFriend", "/searchResults",
+                                "/tripsOwned", "/tripsParticipated", "resources/**", "/downloadGpx", "/delete_participant",
+                                "/findTrip", "/myProfile", "/addRate","/kilometers","/renew_trip",
                                 HttpMethod.POST).fullyAuthenticated()
```
Redundant

Mefjuu94 replied:
right.

blazejprzyluski commented on src/test/java/TripDAOTests.java:
```diff
@@ -0,0 +1,220 @@
+
+@Testcontainers
+public class TripDAOTests {
+    CustomUserDAO customUserDAO;
+    private Trip testTrip = createTestTrip();
+    private final CustomUser customUser = createTestUser();
+    TripDAO testObject = new TripDAO();
+    DockerImageName postgres = DockerImageName.parse("postgres:16");
+    @Container
+    PostgreSQLContainer postgresqlContainer;
```
private

blazejprzyluski commented on src/test/java/TripDAOTests.java:
```diff
@@ -0,0 +1,220 @@
+        testObject = new TripDAO(TestSessionFactoryCreator.getCustomUserSessionFactory(mappedPort));
+    }
+
+    private CustomUser createTestUser() {
+        CustomUser customUser = new CustomUser(1, "test@mail.com", "testUser123!",
+                "testUser", 99, 0, 0, new ArrayList<>(),
+                new ArrayList<>(), "city",0);
+        return customUser;
+    }
+
+    private Trip createTestTrip() {
+        return new Trip.TripBuilder()
+                .withTripDescription("description")
+                .withDestination("destination")
+                .withOwner(customUser)
+                .withTripDuration("5h")
+                .withClosedGroup(false)
+                .withAmountOfClosedGroup(1)
+                .withDriverPeople(false)
+                .withAmountOfDriverPeople(12)
+                .withAnimals(true)
+                .withWaypoints("")
+                .withTripDataTime(LocalDateTime.now())
+                .withDistanceOfTrip("5km")
+                .withGpxFile(null)
+                .build();
+    }
```
Duplicate code

Mefjuu94 replied:
not any more :)

blazejprzyluski commented on src/test/java/TripDAOTests.java:
```diff
@@ -0,0 +1,220 @@
+        // add trip with participants
+        testObject.addAnnouncement(testTrip);
+
+        //now read array of objects (trip and customUser classes)
+        List<Object[]> objects = testObject.listAllTripParticipantIds();
+        List<String> tripsAndIDs = new ArrayList<>();
+
+        for (Object[] obj : objects) {
+            String s1 = Arrays.toString(obj);
+            String s = s1.replaceAll("[\\[\\]\\s]", "");
+            String[] split = s.split(",");
+            String tripID = split[0];
+            String userID = split[1];
+            tripsAndIDs.add(userID + "," + tripID);
+        }
```
duplicate code + unsafe casting

blazejprzyluski commented on src/test/java/UserRatingDAOTests.java:
```diff
@@ -0,0 +1,250 @@
+@Testcontainers
+public class UserRatingDAOTests {
+
+    private CustomUser reviewer = new CustomUser(2, "reviewer@mail.com", "testUser123!",
+            "reviewerTestUser", 99, 0, 0, new ArrayList<>(),
+            new ArrayList<>(), "reviewerCity",0);
+    ///////////////////
+    CustomUserDAO customUserDAO;
+    private final CustomUser customUser = createTestUser();
+    UserRating rate = createRate();
+    private UserRatingDAO testObject;
+    DockerImageName postgres = DockerImageName.parse("postgres:16");
+    private Trip testTrip = createTestTrip();
+    private TripDAO tripDAO = new TripDAO();
+    @Container
+    PostgreSQLContainer postgresqlContainer;
```
private

blazejprzyluski commented on src/test/java/UserRatingDAOTests.java:
```diff
@@ -0,0 +1,250 @@
+        }
+    }
+
+    private Trip createTestTrip() {
+        return new Trip.TripBuilder()
+                .withTripDescription("description")
+                .withDestination("destination")
+                .withOwner(customUser)
+                .withTripDuration("5h")
+                .withClosedGroup(false)
+                .withAmountOfClosedGroup(1)
+                .withDriverPeople(false)
+                .withAmountOfDriverPeople(12)
+                .withAnimals(true)
+                .withWaypoints("")
+                .withTripDataTime(LocalDateTime.now())
+                .withDistanceOfTrip("5km")
+                .withGpxFile(null)
+                .build();
+    }
```
duplicate code

Mefjuu94 replied:
solved

Mefjuu94 committed:
* added support for tests using Singleton - now there is one session instead of a single one for each test - methods responsible for closing the session have been removed
* therefore, all tests were modified where using DAO.
* Added two new classes in tests -> for Singleton purposes.
  1. DAOFactoryForMockTests which pretends to be connected to the database
  2. EntityFactoryForTests which creates entities for these databases.
* the names of some methods have been standardized -> ["/delete_participant"] -> ["/deleteParticipant"]
* changed from RequestMapping to GetMapping some methods as they should be
* changed the dangerous "listOfTrips" method which returned an array of objects which now returns map [user_id],List[trips_id]

Mefjuu94 committed:
* added support for tests using Singleton - now there is one session instead of a single one for each test - methods responsible for closing the session have been removed
* therefore, all tests were modified where using DAO.
* Added two new classes in tests -> for Singleton purposes.
  1. DAOFactoryForMockTests which pretends to be connected to the database
  2. EntityFactoryForTests which creates entities for these databases.
* the names of some methods have been standardized -> ["/delete_participant"] -> ["/deleteParticipant"]
* changed from RequestMapping to GetMapping some methods as they should be
* changed the dangerous "listOfTrips" method which returned an array of objects which now returns map [user_id],List[trips_id]

Mefjuu94 committed:
* added support for tests using Singleton - now there is one session instead of a single one for each test - methods responsible for closing the session have been removed
* therefore, all tests were modified where using DAO.
* Added two new classes in tests -> for Singleton purposes.
  1. DAOFactoryForMockTests which pretends to be connected to the database
  2. EntityFactoryForTests which creates entities for these databases.
* the names of some methods have been standardized -> ["/delete_participant"] -> ["/deleteParticipant"]
* changed from RequestMapping to GetMapping some methods as they should be
* changed the dangerous "listOfTrips" method which returned an array of objects which now returns map [user_id],List[trips_id]

blazejprzyluski commented on src/main/java/TTT/controller/CustomUserController.java:
```diff
@@ -248,30 +114,35 @@ public String updateField(@RequestParam("fieldName") String fieldName,
+
+        Trip trip = tripDAO.findTripID(Long.parseLong(tripId));
+
+        UserRatingDAO userRatingDAO = new UserRatingDAO();
```
Why create DAO here?

Mefjuu94 replied:
right, fixed!

blazejprzyluski commented on src/main/java/TTT/controller/MapController.java:
```diff
@@ -18,59 +15,61 @@
 @Controller
 public class MapController {
 
     TripDAO tripDAO = new TripDAO();
+    MethodsHandler methodsHandler = new MethodsHandler();
```
private final

Mefjuu94 replied:
fixed!

blazejprzyluski commented on src/main/java/TTT/controller/AnnouncementController.java:
```diff
@@ -302,25 +249,45 @@ public String deleteParticipant(@RequestParam String user_Identity,@RequestParam
+
+        Trip trip = tripDAO.findTripID(tripID);
+
+        tripDAO.deleteTrip(trip);
+
+        String nextPage = "/announcement";
+        model.addAttribute("nextPage", nextPage);
+
+        return "actionSuccess";
```
Even if trip deletion fails, user gets success as a response

Mefjuu94 replied:
solved!

blazejprzyluski commented on src/main/java/TTT/controller/MethodsHandler.java:
```diff
@@ -0,0 +1,83 @@
+            break;
+        }
+
+        System.out.println(tripIdsForUser.size());
```
to be deleted

Mefjuu94 replied:
ok!

blazejprzyluski commented on src/main/java/TTT/controller/SecurityController.java:
```diff
@@ -4,55 +4,43 @@
 
-    @RequestMapping(value="/logout", method = RequestMethod.GET)
-    public String logoutPage (HttpServletRequest request, HttpServletResponse response) {
+    @GetMapping(value = "/logout")
```
why explicit use of value parameter?

Mefjuu94 replied:
fixed!

blazejprzyluski commented on src/main/java/TTT/databaseUtils/CommentsDAO.java:
```diff
@@ -104,18 +120,26 @@ public Comments findCommentID(long id) {
     }
 
-    public void editComment(long idOfComment, String commentString) {
+    public boolean editComment(long idOfComment, String commentString) {
```
make sure you use the returned value always

Mefjuu94 committed:
* added proper information about success of action in every controller. Now instead of result;
- success page as always be
- information page with proper information when fail post method.
* added password verification in backend (customUserDao).

Mefjuu94 committed:
* fixed buf in password validation, now everything works fine
* make fields in DAOFactoryForMockTests to be final
* added tests to password verification in CustomUserDAO

blazejprzyluski commented on src/main/java/TTT/controller/PasswordResetController.java:
```diff
@@ -0,0 +1,76 @@
+@Controller
+public class PasswordResetController {
+
+    CustomUserDAO customUserDAO = new CustomUserDAO();
+    CustomUser customUser = new CustomUser();
```
private final

blazejprzyluski commented on src/main/java/TTT/controller/PasswordResetController.java:
```diff
@@ -0,0 +1,76 @@
+    @GetMapping("information")
+    public String getInformationPage(Model model) {
+
+        String information = "please enter valid email or password";
```
could be static variable

blazejprzyluski commented on src/main/java/TTT/controller/PasswordResetController.java:
```diff
@@ -0,0 +1,76 @@
+            }
+        }
+        else {
+            String information = "enter valid EMAIL";
```
could be static variable

blazejprzyluski commented on src/main/java/TTT/databaseUtils/CustomUserDAO.java:
```diff
@@ -11,28 +11,39 @@
 
     private final BCryptPasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
-    private final SessionFactory sessionFactory = UserSessionFactory.getUserSessionFactory();
+    private SessionFactory sessionFactory = UserSessionFactory.getUserSessionFactory();
```
final?

blazejprzyluski commented on src/main/java/TTT/databaseUtils/CustomUserDAO.java:
```diff
@@ -11,28 +11,39 @@
-    private final SessionFactory sessionFactory = UserSessionFactory.getUserSessionFactory();
+    private SessionFactory sessionFactory = UserSessionFactory.getUserSessionFactory();
+
+    public CustomUserDAO() {
+    }
```
why the argless constructor?

blazejprzyluski commented on src/main/java/TTT/databaseUtils/CustomUserDAO.java:
```diff
@@ -201,41 +262,78 @@ public void updateUserAge(String email, int age) {
+        }
+    }
 
+    public boolean isValidPassword(String password) {
```
is this method used outside this class?

blazejprzyluski commented on src/main/java/TTT/databaseUtils/TripDAO.java:
```diff
@@ -1,25 +1,39 @@
+        this.sessionFactory = testSessionFactory;
+    }
+
+    public TripDAO() {
+    }
```
why argless constructor?

blazejprzyluski commented on src/main/java/TTT/security/SecurityConfig.java:
```diff
@@ -32,28 +26,24 @@ public SecurityConfig(CustomUserSecurityService securityService) {
                         .usernameParameter("email")
                         .defaultSuccessUrl("/")
-                        .failureUrl("/login?error=true")
+                        .failureUrl("/information")
```
very generic failure URL

blazejprzyluski commented on src/main/java/TTT/trips/GPX.java:
```diff
@@ -55,15 +55,26 @@ public void makeGPX(String geoJsonString, int numberOfWaypoints, String filePath
+
+        if (!folder.exists()) {
+            if (folder.mkdirs()) {
+                System.out.println("Folder został utworzony: " + folderPath);
+            } else {
+                System.out.println("Nie udało się utworzyć folderu.");
+            }
+        } else {
+            System.out.println("Folder już istnieje.");
```
why polish?

blazejprzyluski commented on src/main/resources/static/js/map.js:
```diff
@@ -1,7 +1,5 @@
-// Replace with your own API key
 const API_KEY = 'zjEFy9NsTMuP_e3U9_B0sDu_axPSSl28smWg1PXW4i0';
```
Might be worth looking into getting this information from a config file

Mefjuu94 committed:
* Now API keys are hidden and loaded from external file "apiKeys.js"
* deleted "information" page, now all errors are maintain from default error pages.

Mefjuu94 committed:
* Now API keys are hidden and loaded from external file "apiKeys.js"
* deleted "information" page, now all errors are maintain from default error pages.

Mefjuu94 committed:
* changed some styles in:
 - rate -> added style that if user can add rate button is green else is gray.
 - trip -> changes style who mutually exclusive self and generates errors
 * Added page to rating of user, now user can see comments with rate.

Mefjuu94 committed:
* fixed some errors that provides wrong redirecting of user
